inference-engine/src/inference_engine/saturated_cast.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <type_traits>
+#include <limits>
+#include <algorithm>
+
+namespace InferenceEngine {
+
+template <typename OutT, typename InT>
+inline typename std::enable_if<
+        std::is_integral<OutT>::value && std::is_integral<InT>::value &&
+        std::is_signed<InT>::value &&
+        !std::is_same<OutT, InT>::value,
+        OutT>::type saturated_cast(InT value) {
```

Comment:
Is it possible to use `using` here to avoid long return type?
Is it possible to move `std::enable_of` in `template<...>` part to separate output type and predicate?

Reply:
Nitpick: `class` is shorter than `typename`.. :)

Reply:
I would have an aligned with OpenCV name `saturate_cast`

Reply:
I don't think it's possible to move `std::enable_if` because for this predicate we need both `InT` and `OutT` to be declared, so I haven't found possibility to insert it there.
Done: `typename` -> `class`, `saturated_cast` -> `saturate_cast`

Reply:
What is about

```cpp
template <class To, class From, typename = std::enable_if<
    !std::is_same<To, From>::value &&
    std::is_integral<To>::value &&
    std::is_integral<From>::value &&
    std::is_signed<From>::value
>::type>
To saturate_cast(From const& value);
```

Reply:
I have finally found the working one, thanks!